tests/security/389ds/tls_389ds_sssd_client.pm

```diff
@@ -0,0 +1,77 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Implement & Integrate 389ds + sssd test case into openQA,
+#          This test module covers the sssd client tests
+#
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#88513
+
+use base 'consoletest';
+use testapi;
+use strict;
+use warnings;
+use utils;
+use lockapi;
+use Utils::Systemd qw(systemctl disable_and_stop_service);
+
+sub run {
+    select_console("root-console");
+
+    my $local_ip    = '10.0.2.102';
+    my $remote_ip   = '10.0.2.101';
+    my $local_name  = '389dc';
+    my $remote_name = '389ds';
+    my $inst_ca_dir = '/etc/dirsrv/slapd-localhost';
+    my $tls_dir     = '/etc/openldap/certs';
+    my $ldap_user   = 'wilber';
+    my $uid         = '1003';
+
+    # Install 389-ds and sssd on client
+    zypper_call("in 389-ds sssd");
+
+    # Disable teh nscd daemon
+    disable_and_stop_service("nscd", ignore_failure => 1);
+
+    # Configure host names for C/S communication
+    assert_script_run("sed -i '/$local_ip.*/d' /etc/hosts");
+    assert_script_run("sed -i '/$remote_ip.*/d' /etc/hosts");
+    assert_script_run("echo '$local_ip $local_name.example.com $local_name' >> /etc/hosts");
+    assert_script_run("echo '$remote_ip $remote_name.example.com $remote_name' >> /etc/hosts");
+    assert_script_run("hostname $local_name");
+
+    # Verify sssd authentication on sssd client
+    mutex_wait("389DS_READY");
+    assert_script_run("mkdir -p $tls_dir");
+    exec_and_insert_password("scp -o StrictHostKeyChecking=no root\@$remote_ip:/tmp/sssd.conf /tmp");
+    exec_and_insert_password("scp -o StrictHostKeyChecking=no root\@$remote_ip:$inst_ca_dir/ca.crt $tls_dir");
+    assert_script_run("/usr/bin/openssl rehash $tls_dir");
+    assert_script_run("cat /tmp/sssd.conf > /etc/sssd/sssd.conf");
+    assert_script_run("sed -i '/^passwd.*/ s/\$/ sss/' /etc/nsswitch.conf");
+    assert_script_run("sed -i '/^group.*/ s/\$/ sss/' /etc/nsswitch.conf");
+    assert_script_run("sed -i '/^shadow.*/ s/\$/ sss/' /etc/nsswitch.conf");
```

Comment:
Nice comments, I was willing to do this, however, based on the comments from Fabian:
"A variable only really makes sense if it is either variable (i.e. can change) or is shorter than its content. Here neither is the case, so I suggest to just remove it."
It is hard for me to use a better variable than use default system config file there. so just re-used this default name.

Reply:
> Nice comments, I was willing to do this, however, based on the comments from Fabian:
> "A variable only really makes sense if it is either variable (i.e. can change) or is shorter than its content. Here neither is the case, so I suggest to just remove it."
> It is hard for me to use a better variable than use default system config file there. so just re-used this default name.

It is up to you I agree with @Vogtinator but I do not think @Vogtinator means this case.

Reply:
Thanks Lili! and you are right, I don't know if I understand right for Fabian's comments. let me work it out before revising the code

@Vogtinator Can I ask for your kindly help to take a look at this?

Reply:
Using a variable here doesn't make much sense, as it will always have a constant value everywhere it only makes it harder to read with no advantage.

However, this snippet is broken on Tumbleweed, where it's necessary to copy `/usr/etc/nsswitch.conf` to `/etc/nsswitch.conf` before any changes can be done.

Reply:
What you could do is merge those sed calls all together, as `sed` allows running multiple commands at once.

Reply:
Thanks much! yes, one single command is better, let me fix it.
I just triggered the tests on TW, and glad to know that it uses different nsswitch.conf file.